BugSplatDotNetStandard/Api/BugSplatApiClient.cs

```diff
@@ -0,0 +1,127 @@
+        /// </summary>
+        /// <param name="route">Relative URL that is relative to Host</param>
+        /// <param name="content">POST content</param>
+        Task<HttpResponseMessage> PostAsync(string route, HttpContent content);
+
+        /// <summary>
+        /// Make a GET request to a route relative to Host
+        /// </summary>
+        /// <param name="route">Relative URL that is relative to Host</param>
+        Task<HttpResponseMessage> GetAsync(string route);
+    }
+
+    /// <summary>
+    /// Used to make authenticated request to various BugSplat APIs via email and password authentication
+    /// </summary>
+    public class BugSplatApiClient : IBugSplatApiClient
+    {
+        public bool Authenticated { get; private set; }
+
+        public Uri Host { get; set; } = new Uri("https://app.bugsplat.com");
+
+        private string email;
+        private string password;
+        private HttpClient httpClient;
+
+        internal BugSplatApiClient(
+            string email,
+            string password,
+            IHttpClientFactory httpClientFactory
+        )
+        {
+            ThrowIfArgumentIsNullOrEmpty(email, "email");
+            ThrowIfArgumentIsNullOrEmpty(password, "password");
+            ThrowIfArgumentIsNull(httpClientFactory, "httpClientFactory");
+
+            this.email = email;
+            this.password = password;
+            this.httpClient = httpClientFactory.CreateClient();
+        }
+
+        /// <summary>
+        /// Create an unauthenticated BugSplatApiClient that will use an email and password to authenticate
+        /// </summary>
+        /// <param name="email">BugSplat account email</param>
+        /// <param name="password">BugSplat account password</param>
+        public static BugSplatApiClient Create(string email, string password)
+        {
+            return new BugSplatApiClient(email, password, HttpClientFactory.Default);
+        }
+
+        /// <summary>
+        /// Authenticate with the BugSplat API via email and password
+        /// </summary>
+        public async Task<HttpResponseMessage> Authenticate()
+        {
+            var url = new Uri(Host, "/api/authenticatev3");
+            var formData = new MultipartFormDataContent()
+            {
+                { new StringContent(email), "email" },
+                { new StringContent(password), "password" }
+            };
+
+            var authenticateResponse = await this.httpClient.PostAsync(url, formData);
+
+            ThrowIfHttpRequestFailed(authenticateResponse);
+
+            var setCookieHeader = authenticateResponse.Headers.FirstOrDefault(header => header.Key.Contains("Set-Cookie")).Value;
+            var xsrfCookie = setCookieHeader.FirstOrDefault(value => value.Contains("xsrf-token"));
+            var xsrfToken = xsrfCookie.Split(';').First().Split('=').Last();
+            this.httpClient.DefaultRequestHeaders.Add("xsrf-token", xsrfToken);
```

Comment:
Is there a good reason to do xsrf-token as a header on POST only? Are there security implications associated with it being on a GET?

Reply:
this seems fine according to https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html and doing it this way allows us to add support for PUT and DELETE with no extra work code.